widgetssdk/src/main/java/com/glia/widgets/chat/adapter/CustomMessageAdapter.java

```diff
@@ -0,0 +1,66 @@
+package com.glia.widgets.chat.adapter;
+
+import static com.glia.widgets.chat.adapter.ChatAdapter.CUSTOM_MESSAGE_TYPE;
+
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+
+import com.glia.androidsdk.chat.ChatMessage;
+import com.glia.widgets.UiTheme;
+import com.glia.widgets.chat.adapter.holder.CustomMessageViewHolder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class CustomMessageAdapter {
+    @VisibleForTesting
+    final Map<Integer, Integer> viewTypeMap = new HashMap<>();
+
+    @Nullable
+    abstract public Integer getItemViewType(ChatMessage message);
```

Comment:
No, this is `@Nullable` value. I will add the JavaDoc for it. `null` means this `ChatMessage` is not a custom view type.

Reply:
Just a thought, as I don't know all preconditions: maybe it makes sense to return itemViewType for all messages (including some value for default messages)? Because it looks a bit confusing to me that default messages won't have a view type.

Reply:
Guys, this Adapter is only for Custom messages(`metadata != null`). All other message types are provided by ChatAdapter and Integrator can't affect for them.
I hope I can describe that all in JavaDoc and Documentation :)